### DISCUSSION TITLE
Developing formulae or casks

### DISCUSSION THREAD

**Discussion by austek** [Homebrew/discussions#5036] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.0-19-g23ce2d5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 23ce2d5bf81e71d8bf9fdf9198d8d23724f0966b
Last commit: 3 weeks ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 607778083a4383ab97b6ffc719a11a79f875e1cc
Core tap last commit: 4 weeks ago
Core tap branch: espeak-ng_1.51.1
Core tap JSON: 19 Dec 18:36 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: ["--appdir=/Users/ali.ustek/Applications"]
HOMEBREW_EDITOR: code
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/blkid.pc
  /usr/local/lib/pkgconfig/com_err.pc
  /usr/local/lib/pkgconfig/e2p.pc
  /usr/local/lib/pkgconfig/ext2fs.pc
  /usr/local/lib/pkgconfig/fuse-ext2.pc
  /usr/local/lib/pkgconfig/ss.pc
  /usr/local/lib/pkgconfig/uuid.pc

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo austek/tap) checkout
  git -C $(brew --repo homebrew/core) checkout master
```

### Description of issue

I've created a PR https://github.com/Homebrew/homebrew-core/pull/157428 and need help/guidance with the failures

Down to 3 failures now, 

1. `GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)`
  this is for `pcaudiolib` which is a required dependency for `espeak-ng`
2. `Formulae should not require patches to build. Patches should be submitted and accepted upstream first.`
Upstream is fixed but not released yet 
https://github.com/espeak-ng/espeak-ng/pull/1688
3. `Undeclared dependencies with linkage: pcaudiolib`
I don't know what to do with this one?

**Comment by SMillerDev** [comment]
> this is for pcaudiolib which is a required dependency for espeak-ng

Preferably we would have the dependency gain some popularity first, but this can be waved by maintainers if needed.
> Upstream is fixed but not released yet

So we should wait for that release 
> Undeclared dependencies with linkage: pcaudiolib

Your formula should declare a dependency on pcaudiolib

**Reply by austek** [reply]
> Your formula should declare a dependency on pcaudiolib

I thought I had declared in https://github.com/Homebrew/homebrew-core/blob/607778083a4383ab97b6ffc719a11a79f875e1cc/Formula/e/espeak-ng.rb#L12

**Reply by SMillerDev** [reply]
That only declares a build time dependency, while the audit indicates that it also depends on it at runtime
